pkg/queue/queue.go

```diff
@@ -213,3 +222,49 @@ func (h *heapImpl) Pop() interface{} {
 	delete(h.items, key)
 	return obj
 }
+
+// UnschedulableWorkloadsMap implementation inspired by
+// https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/internal/queue/queue.go
+
+// UnschedulableWorkloadsMap holds workloads that cannot be scheduled.
+// This data structure is used to implement unschedulableQ.
+type UnschedulableWorkloadsMap struct {
+	// workloadInfoMap is a map key by a workload's full-name and the value
+	// is a/pointer to the workload.Info.
+	workloadInfoMap map[string]*workload.Info
+}
+
+// Add adds a workload to the unschedulable workloadInfoMap.
+func (u *UnschedulableWorkloadsMap) addOrUpdate(wInfo *workload.Info) {
+	key := workload.Key(wInfo.Obj)
+	u.workloadInfoMap[key] = wInfo
+}
+
+// Delete deletes a workload from the unschedulable workloadInfoMap.
+func (u *UnschedulableWorkloadsMap) delete(w *kueue.QueuedWorkload) {
+	key := workload.Key(w)
+	delete(u.workloadInfoMap, key)
+}
+
+// TODO unused now
```

Comment:
non of the others are used as well :)

I suggest we either uncomment and unit test it or remove this code.

Reply:
Done